### ISSUE TITLE
Validating credit card expiration date using two separate fields

### ISSUE THREAD

**Issue by laxman4434** [formvalidation/support#439]
Hi nghuuphuoc,

http://formvalidation.io/examples/validating-credit-card-expiration-date/

While using credit card expiration date using two separate fields, is it possible to allow only last two digits in the year field instead of four digits

Thanks in advance

**Comment by nghuuphuoc** [comment]
You can use the `transformer` option. The idea is that using this option to transform a two digits year number into a four digits year number:

```javascript
$(document).ready(function() {
    $('#paymentForm').formValidation({
        framework: 'bootstrap',
        icon: {
            ...
        },
        fields: {
            expMonth: {
                row: '.col-xs-3',
                validators: {
                    // ... The same code as seen in the example
                }
            },
            expYear: {
                row: '.col-xs-3',
                // HERE is the new code
                transformer: function($field, validatorName, validator) {
                	var year = parseInt($field.val(), 10);
                  if (isNaN(year)) {
                  	return year;
                  } else {
                  	// Add 2000 to year
                  	return 2000 + year;
                  }
                },
                validators: {
                    // ... The same code as seen in the example
                }
            }
        }
    });
});
```

I made a fiddle for this at http://jsfiddle.net/nghuuphuoc/juavhwo0/2/

![credit-card](https://cloud.githubusercontent.com/assets/4226036/12037886/c122f510-ae85-11e5-834c-cfecae15d35f.gif)

__IMPORTANT__
You should take a look at the [Modifying the value before validating](http://formvalidation.io/examples/modifying-value-before-validating/) example to see how powerful the `transformer` option is.

**Comment by laxman4434** [comment]
Thanks for your response nghuuphuoc.

I know it's working as you have shown in JSFiddle.
But for some reason, the above method is throwing an error for me.

'''
0x800a01b6 - JavaScript runtime error: Object doesn't support property or method 'replace'
'''
in formValidation.min.js

But i tried another way.

'''
callback: function (value, validator, $field) {
                               value = "20" + value;     //added this line
                               value = parseInt(value, 10);
                               var month = validator.getFieldElements('ctl00$htp_cph$Wizard_CheckOut$uctrl_payment$txt_PY_Month').val(),
                                   currentMonth = new Date().getMonth() + 1,
                                   currentYear = new Date().getFullYear();
                               if (value < currentYear || value > currentYear + 10) {
                                   return false;
                               }
                               if (month == '') {
                                   return false;
                               }
                               month = parseInt(month, 10);
                               if (value > currentYear || (value == currentYear && month >= currentMonth)) {
                                   validator.updateStatus('ctl00$htp_cph$Wizard_CheckOut$uctrl_payment$txt_PY_Month', 'VALID');
                                   return true;
                               } else {
                                   return false;
                               }
                           }
'''

**Comment by nghuuphuoc** [comment]
Modifying `value = "20" + value;` directly might not be the best solution.
Does the error `0x800a01b6 - JavaScript runtime error: Object doesn't support property or method 'replace'` happen on other browsers such as Chrome or Firefox?

**Comment by laxman4434** [comment]
Ok I found out what the problem is.

```
<input  class="form-control" placeholder="Year (YY)" name="expYear" type="number" pattern="[0-9]*" inputmode="numeric" />
```
I have added HTML5 attributes to my input to show numbers keyboard on mobiles
```
 type="number" pattern="[0-9]*" inputmode="numeric"
```

When I tried to debug the javascript, I found out that the transformer function validates the validators 

"notEmpty","digits","callback" and then also "number", "regexp".

When validating the "number" validator, error is being thrown.

When i remove the type="number" attribute, everything works fine and all the four validators

"notEmpty","digits","callback"  and "regexp" are being validated properly.
